libvast/src/system/local_segment_store.cpp

```diff
@@ -308,7 +308,7 @@ class local_store_plugin final : public virtual store_plugin {
   }
 
   [[nodiscard]] const char* name() const override {
-    return "local-segment-store";
+    return "segment-store";
```

Comment:
It's a bit weird that the plugin name differs from the class and file names, why is that?

Reply:
We thought that `local-segment-store` is too technical for a user-facing name, so switched it to just "segment-store". But there already exist files called `segment_store.{cpp,hpp}`, so I thought since the file name is developer-facing it should be fine to keep it as `local_segment_store.hpp` so its easy to remember whats the difference between this and the other file.

Of course, naming is the hardest part, so feel free to suggest a better alternative name.

Reply:
I have no hard opinion on this, was just a bit confused. Thanks for explaining.

For naming suggestions: Isn't `segment` already too technical? You could call it just `store`, or `file-store`. But then again this really is bikeshedding.

Reply:
Just `store` sounds a bit too generic if we assume that we might have different store backends at some point. Using `file-store` sounds fine to me, personally I'm also starting to like `local-store` (saying that the data is stored on your local hard drive)

Reply:
👍 for `local-store` from my side. @tobim what do you think?

Reply:
How about `event-store`?

Reply:
I don't think `event-store` conveys that it is local to the database.

Reply:
From a user's perspective, isn't this just a `partition-store` plugin? At least high-level, the developer/user thinks of the block of events as a (horizontal) partition, so the way the data is stored would naturally be called the `partition-store`.

Reply:
If I understand you correctly you're saying that `partition-store` would be a more intuitive name for the "category" of store plugins? (ie. what is currently called `store-backend` in this PR) That still leaves the question how to name this particular implementation of a "partition store", which is what we originally discussed here.

Reply:
We should look at these names based on the context of what other implementations we might have in the future. Most of those have obvious names. I'm thinking of `parquet-store` of `s3-store`. In that context we can rule out some options thanks to ambiguities: `file-store` is in conflict with `parquet-store`, because parquet is a file too. `local-store` also conflicts with `parquet-store` because both are local. It seems that single properties don't serve as good names here.
I'm equally unenthusiastic about `event-store` and `partition-store` because every implementation will store events at the partition granularity.

From the options we have at the moment `segment-store` seems to be the best, even though the name is not descriptive for operators without implementation knowledge. The big advantage is that it isn't misleading. The opaque name will hopefully prompt users to read the documentation instead of going with some assumptions.

Reply:
I'll go with `segment-store` for now to get the PR merged; the plugin is still clearly labelled as experimental so there should be no issue renaming this later on if so desired.